src/main/core/worker.rs

```diff
@@ -1,188 +1,177 @@
+use nix::unistd::Pid;
+use once_cell::unsync::OnceCell;
+
 use crate::core::support::simulation_time::SimulationTime;
 use crate::cshadow;
 use crate::host::host::Host;
-use crate::host::memory_manager::MemoryManager;
+use crate::host::host::HostInfo;
 use crate::host::process::Process;
+use crate::host::process::ProcessId;
+use crate::host::thread::ThreadId;
 use crate::host::thread::{CThread, Thread};
 use crate::utility::notnull::*;
 use std::cell::RefCell;
+use std::sync::Arc;
 
 #[derive(Copy, Clone, Debug)]
 pub struct WorkerThreadID(u32);
 
+struct ActiveProcessInfo {
+    #[allow(dead_code)]
+    id: ProcessId,
+    native_pid: Pid,
+}
+
+struct ActiveThreadInfo {
+    #[allow(dead_code)]
+    id: ThreadId,
+    native_tid: Pid,
+}
+
 /// Worker context, capturing e.g. the current Process and Thread.
 pub struct Worker {
```

Comment:
Is this comment still true? It seems like now the Worker is just a global cache for information about the current host, process, and thread, but not the host, process, or thread themselves (except for the legacy `c_` pointers)?